homeassistant/components/homeassistant_hardware/update.py

```diff
@@ -0,0 +1,331 @@
+        raise NotImplementedError
+
+    def _maybe_recompute_state(self) -> None:
+        """Recompute the state of the entity."""
+
+        if self._current_firmware_info is None:
+            firmware_type = None
+        else:
+            firmware_type = self._current_firmware_info.firmware_type
+
+        if firmware_type not in self.firmware_entity_descriptions:
+            _LOGGER.warning(
+                "Unexpected firmware type %r, assuming Zigbee firmware instead",
+                firmware_type,
+            )
+            firmware_type = ApplicationType.EZSP
+
+        self.entity_description = self.firmware_entity_descriptions[firmware_type]
+        if self._latest_manifest is None:
+            return
+
+        self._latest_firmware = next(
+            f
+            for f in self._latest_manifest.firmwares
+            if f.filename.startswith(self.entity_description.fw_type)
+        )
+
+        version = cast(
+            str, self._latest_firmware.metadata[self.entity_description.version_key]
+        )
+        self._attr_latest_version = self.entity_description.version_parser(version)
+        self._attr_release_summary = self._latest_firmware.release_notes
+        self._attr_release_url = str(self._latest_manifest.html_url)
+
+        firmware_name = self.entity_description.firmware_name
+
+        if (
+            self._current_firmware_info is None
+            or self._current_firmware_info.firmware_version is None
+        ):
+            sw_version = None
+        else:
+            firmware_version = self.entity_description.version_parser(
+                self._current_firmware_info.firmware_version
+            )
+            sw_version = f"{firmware_name} {firmware_version}"
+
+        if self.device_entry is not None:
+            dr.async_get(self.hass).async_update_device(
+                self.device_entry.id, sw_version=sw_version
+            )
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._latest_manifest = self.coordinator.data
+        self._maybe_recompute_state()
+        self.async_write_ha_state()
+
+    @property
+    def installed_version(self) -> str | None:
+        """Version installed and in use."""
+        if self._current_firmware_info is None:
+            return None
+
+        if (version := self._current_firmware_info.firmware_version) is None:
+            return None
+
+        return self.entity_description.version_parser(version)
+
+    def _update_progress(self, offset: int, total_size: int) -> None:
+        """Handle update progress."""
+        self._attr_update_percentage = (offset * 100) / total_size
+        self.async_write_ha_state()
+
+    @asynccontextmanager
+    async def _temporarily_stop_owning_software(
+        self, device: str
+    ) -> AsyncIterator[None]:
+        """Temporarily stop owning addons and integrations."""
```

Comment:
I think we need a longer description what it means to stop owning the software.

Reply:
Integrations and addons that are currently using the device and running will be stopped by the context manager and then restarted once it exits.

Reply:
Please add that to the docstring. It clarifies that it's the add-ons and integrations that are owning something, not that this method is stopping owning something.